spec/api.spec.ts

```diff
@@ -216,3 +216,31 @@ describe("isReadOnlyNode", () => {
     expect(isReadOnlyNode(types.map(types.string).createReadOnly({ foo: "bar" }))).toEqual(true);
   });
 });
+
+describe("resolveIdentifier", () => {
+  describe.each([
+    ["readonly", true],
+    ["mutable", false],
+  ])("on %s nodes", (_name, readonly) => {
+    let instance: TestClassModel;
+    beforeEach(() => {
+      instance = readonly ? TestClassModel.createReadOnly(TestModelSnapshot) : TestClassModel.create(TestModelSnapshot);
+    });
+
+    test("resolves an identifier from the root of the tree", () => {
+      expect(resolveIdentifier(NamedThingClass, instance, "mixed_up")).toBe(instance.nested);
+
+      expect(resolveIdentifier(NamedThingClass, instance, "mixed_up")?.key).toEqual("mixed_up");
+      expect(resolveIdentifier(NamedThingClass, instance, "test_key")?.key).toEqual("test_key");
+    });
+
+    test("resolves an identifier from within the tree", () => {
+      expect(resolveIdentifier(NamedThingClass, instance.array[0], "mixed_up")?.key).toEqual("mixed_up");
+      expect(resolveIdentifier(NamedThingClass, instance.map.get("test_key")!, "test_key")?.key).toEqual("test_key");
+    });
+
+    test("returns null when the identifier doesn't resolve", () => {
```

Comment:
super nit: `returns undefined` (which is good too, because that's what MST does!)